apiserver/facades/client/applicationoffers/applicationoffers.go

```diff
@@ -43,28 +48,29 @@ func createOffersAPI(
 	authorizer facade.Authorizer,
 	resources facade.Resources,
 	authContext *commoncrossmodel.AuthContext,
-) (*OffersAPI, error) {
+) (*OffersAPIV2, error) {
 	if !authorizer.AuthClient() {
 		return nil, common.ErrPerm
 	}
 
 	dataDir := resources.Get("dataDir").(common.StringResource)
-	api := &OffersAPI{
-		dataDir:      dataDir.String(),
-		authContext:  authContext,
-		APIAddresser: common.NewAPIAddresser(backend.GetAddressAndCertGetter(), resources),
-		BaseAPI: BaseAPI{
-			Authorizer:           authorizer,
-			GetApplicationOffers: getApplicationOffers,
-			ControllerModel:      backend,
-			StatePool:            statePool,
-			getEnviron:           getEnviron,
-		}}
+	api := &OffersAPIV2{
+		OffersAPI: &OffersAPI{
```

Comment:
I think you'd want to do all this in NewOffersAPI(ctx facade.Context) (*OffersAPI, error) and just call it here.
This does not look right as-is atm...

Reply:
I reworked it to fix the bad V1 facade